### ISSUE TITLE
Enable Fused AdamW Optimizer

### ISSUE THREAD

**Issue by tokenbender** [tokenbender/avataRL#31]
## Description
Auto-detect and use fused AdamW kernels when available for 10-20% optimizer speedup.

## Tasks
- [ ] Check for fused kernel availability
- [ ] Update optimizer initialization to use fused=True when available
- [ ] Add logging to show if fused optimizer is being used
- [ ] Ensure compatibility with gradient accumulation

## Priority
Medium - Performance optimization

## Technical Details
- Check availability:
```python
import inspect
fused_available = 'fused' in inspect.signature(torch.optim.AdamW).parameters
use_fused = fused_available and torch.cuda.is_available()
```
- Pass fused=True to AdamW when available
- Works on modern GPUs with PyTorch 2.0+
